include/revng/Model/Qualifier.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include "revng/Model/GeneratedQualifier.h"
+
+/// \brief A qualifier for a model::Type
+class model::Qualifier : public model::generated::Qualifier {
+public:
+  using generated::Qualifier::Qualifier;
+
+public:
+  // Kind is not Invalid, Pointer and Const have no Size, Array has Size.
+  bool verify() const debug_function;
+  bool verify(bool Assert) const debug_function;
+  bool verify(VerifyHelper &VH) const;
+  void dump() const debug_function;
+
+public:
+  static Qualifier createConst() { return Qualifier(QualifierKind::Const, 0); }
+
+  static Qualifier createPointer(uint64_t Size) {
+    return Qualifier(QualifierKind::Pointer, Size);
+  }
+
+  static Qualifier createArray(uint64_t S) {
+    return Qualifier(QualifierKind::Array, S);
+  }
+
+public:
+  bool isConstQualifier() const {
+    revng_assert(verify(true));
+    return Kind == QualifierKind::Const;
+  }
+
+  bool isArrayQualifier() const {
+    revng_assert(verify(true));
+    return Kind == QualifierKind::Array;
+  }
+
+  bool isPointerQualifier() const {
+    revng_assert(verify(true));
+    return Kind == QualifierKind::Pointer;
+  }
+};
+
+// Make std::vector<model::Qualifier> yaml-serializable as a sequence
+LLVM_YAML_IS_SEQUENCE_VECTOR(model::Qualifier)
```

Comment:
Let's push this in generated code, if we want to customize this we can add an option in the YAML.